pprof: add new page (#3450)
tldr-pages/tldr: pages/common/pprof.md

# pprof

> > Comand-line tool for visualization and analysis of profiling data.

Reviewer: typo at Comand. And also "analysis of profile data" sounds better.

> > Command-line tool for visualization and analysis analysis of profile data.

Reviewer 2: Double use of `analysis`. Maybe this is what you meant?
```suggestion
> Command-line tool for visualization and analysis of profile data.
```

Author: thanks, I made a commit with this fix

> More information: <https://github.com/google/pprof>.

> - Generate a text report from `fibbo-profile.pb.gz` file, on fibbo binary:

Reviewer 3: Perhaps this will read better?
```suggestion
- Generate a text report from a specific file file, on fibbo binary:
```

What does the "on fibbo binary" bit here mean please? I'm a bit confused.

> - Generate a text report from fibbo-profile.pb.gz profile data on fibbo binary:

Reviewer: fibbo-profile.pb.gz within backticks here and after. And no need for "profile data". It is understood with the name that it is a profile.

`pprof -top {{./fibbo}} {{./fibbo-profile.pb.gz}}`

> - Generate a graph from fibbo-profile.pb.gz profile data and open it on a web browser:

Reviewer 3: Perhaps this will read better?
```suggestion
- Generate a graph and open it on a web browser:
```

We've already used similar language in the previous example, so I'm not sure we need to repeat ourselves here.

`pprof -svg {{./fibbo}} {{./fibbo-profile.pb.gz}}`

> - Run pprof on interactive mode and customize your report with pprof command:

Reviewer: I don't understand what does "customize your report with the pprof command" mean ? Can you clarify ?

> - Run pprof on interactive mode to be able to manually launch pprof command on your file:

Reviewer 3: Perhaps this will read better?
```suggestion
- Run pprof in interactive mode to manually launch `pprof` on a file:
```

`pprof {{./fibbo}} {{./fibbo-profile.pb.gz}}`

> - Run a web server to have a web interface on top of pprof:

Reviewer 3: Perhaps this will read better?
```suggestion
- Run a web server that serves a web interface on top of `pprof`:
```

`pprof -http={{localhost:8080}} {{./fibbo}} {{./fibbo-profile.pb.gz}}`

- Fetch a profile from an HTTP server and generate a report:

> `pprof {{http://localhost:8080/debug/pprof/profile}}`

Reviewer: That endpoint is Go specific. I would rewrite it to be a generic one like - /pprof
